### PR TITLE
Move computing pod targets by config to the analyzer

### PR DESCRIPTION
**Pull request by segiddins**
For performance, since the analyzer has the dependency cache

- [x] CHANGELOG
- [x] Docs
- [x] Move deleted tests

### CODE REVIEW

File: lib/cocoapods/installer/analyzer.rb [hunk 1]
```diff
@@ -465,13 +466,45 @@ def generate_target(target_definition, pod_targets, resolver_specs_by_target)
       # @param  [Hash{Podfile::TargetDefinition => Array<ResolvedSpecification>}] resolver_specs_by_target
       #         the resolved specifications grouped by target.
       #
-      # @return [Array<PodTarget>] the filtered list of pod targets.
+      # @param  [Array<String>] build_configurations
+      #         XXXX
```
Comment (segiddins) [inline lib/cocoapods/installer/analyzer.rb]: Add docs

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* [WIP] Move computing pod targets by config to the analyzer  
  [segiddins](https://github.com/segiddins)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -465,13 +466,45 @@ def generate_target(target_definition, pod_targets, resolver_specs_by_target)
...
       #
-      def filter_pod_targets_for_target_definition(target_definition, pod_targets, resolver_specs_by_target)
-        pod_targets.select do |pod_target|
-          next false unless pod_target.target_definitions.include?(target_definition)
-          resolver_specs_by_target[target_definition].any? { |resolver_spec| !resolver_spec.used_by_tests_only? && pod_target.specs.include?(resolver_spec.spec) }
+      # @return [Hash<String => Array<PodTarget><] the filtered list of pod targets.
```
Comment (segiddins) [inline lib/cocoapods/installer/analyzer.rb]: update

File: lib/cocoapods/target/pod_target.rb [hunk 3]
```diff
@@ -594,21 +563,5 @@ def uses_modular_headers?(only_if_defines_modules = true)
...
-    # @param  [TargetDefinition] target_definition
-    #         The target definition to check.
-    #
-    # @return [Array<Dependency>] The dependency of the target definition for
-    #         this Pod. Return an empty array if the Pod is not a direct
-    #         dependency of the target definition but the dependency of one or
-    #         more Pods.
-    #
-    def target_definition_dependencies(target_definition)
-      target_definition.dependencies.select do |dependency|
```
Comment (segiddins) [inline lib/cocoapods/target/pod_target.rb]: This is the slow bit that's being removed -- `target_definition.dependencies` isn't cheap

File: spec/unit/target/pod_target_spec.rb [hunk 4]
```diff
@@ -71,28 +71,6 @@ module Pod
         @pod_target.dependencies.should == ['monkey']
       end
 
-      it 'returns whether it is whitelisted in a build configuration' do
```
Comment (segiddins) [inline spec/unit/target/pod_target_spec.rb]: move these tests to the analyzer now

Comment (segiddins) [reply spec/unit/target/pod_target_spec.rb]: moved

**Pull request merged**